projects/igniteui-angular/src/lib/accordion/accordion.component.ts

```diff
@@ -0,0 +1,276 @@
+import { IgxExpansionPanelComponent } from '../expansion-panel/expansion-panel.component';
+import { ToggleAnimationSettings } from '../expansion-panel/toggle-animation-component';
+import { IgxAccordionExpansionMode } from './accordion.common';
+import { IgxAccordionService } from './accordion.service';
+
+let NEXT_ID = 0;
+
+@Component({
+    selector: 'igx-accordion',
+    templateUrl: 'accordion.component.html',
+    providers: [IgxAccordionService]
+})
+export class IgxAccordionComponent implements AfterContentInit, AfterViewInit, OnDestroy {
+    /**
+     * Sets/gets the `id` of the accordion component.
+     * If not set, `id` will have value `"igx-accordion-0"`;
+     * ```html
+     * <igx-accordion id = "my-first-accordion"></igx-expansion-panel>
+     * ```
+     * ```typescript
+     * const accordionId =  this.accordion.id;
+     * ```
+     *
+     * @memberof IgxAccordionComponent
+     */
+    @HostBinding('attr.id')
+    @Input()
+    public id = `igx-accordion-${NEXT_ID++}`;
+
+    /** Get/Set the animation settings that panels should use when expanding/collpasing.
+     *
+     * ```html
+     * <igx-accordion [animationSettings]="customAnimationSettings"></igx-accordion>
+     * ```
+     *
+     * ```typescript
+     * const animationSettings: ToggleAnimationSettings = {
+     *      openAnimation: growVerIn,
+     *      closeAnimation: growVerOut
+     * };
+     *
+     * this.accordion.animationSettings = animationSettings;
+     * ```
+     */
+    @Input()
+    public get animationSettings(): ToggleAnimationSettings {
+        return this._animationSettings;
+    }
+    public set animationSettings(value: ToggleAnimationSettings) {
+        this._animationSettings = value;
+        this.updatePanelsAnimation();
+    }
+
+    /**
+     * Gets/Sets accordion expansion mode
+     *
+     * @remarks
+     * By default the accordion expansion mode is 'Single'
+     * @param expansionMode: IgxAccordionExpansionMode
+     */
+    @Input()
+    public get expansionMode() {
+        return this._expansionMode;
+    }
+    public set expansionMode(expansionMode: IgxAccordionExpansionMode) {
+        if (expansionMode === IgxAccordionExpansionMode.Single && this.expansionMode === IgxAccordionExpansionMode.Multiple) {
+            const lastExpanded = this.panels.filter(p => !p.collapsed).pop();
+            this.accordionService.expandedPanels.forEach(panel => {
+                if (panel !== lastExpanded) {
+                    panel.collapse();
+                }
+            });
+        }
+        this._expansionMode = expansionMode;
+    }
+
+    /** Emitted when a panel is expanding, before it finishes
+     *
+     * ```html
+     * <igx-accordion (panelExpanding)="handlePanelExpanding($event)">
+     * </igx-accordion>
+     * ```
+     *
+     *```typescript
+     * public handlePanelExpanding(event) {
```

Comment:
Type the event in the example handler
```suggestion
     * public handlePanelExpanding(event: IExpansionPanelCancelableEventArgs) {
```